### PR TITLE
Deprecate oc secrets subcomands

### PR DESCRIPTION
**Pull request by soltysh**
While recently working on the secrets issue I've noticed we have a set of `oc secrets` subcommands that can be easily replaced with `oc create secret` command. Thus, after discussing this topic with @juanvallejo and @deads2k I'm deprecating them.

/assign @juanvallejo @deads2k

### CODE REVIEW

**Comment by mfojtik** [comment]
LGTM

You will need to update the release notes for 3.9 and mark them as deprecated and being removed in 3.10. The release note issue should be in openshift-docs.

**Comment by soltysh** [comment]
> You will need to update the release notes for 3.9 and mark them as deprecated and being removed in 3.10. The release note issue should be in openshift-docs.

Once this merges I'll update the docs. We are using this in several places and that needs to be changed as well.

File: pkg/oc/cli/secrets/subcommand.go [hunk 1]
```diff
@@ -44,11 +44,6 @@ func NewCmdSecrets(name, fullName string, f *clientcmd.Factory, reader io.Reader
 		Run:     cmdutil.DefaultSubCommandRun(errOut),
 	}
 
-	newSecretFullName := fullName + " " + NewSecretRecommendedCommandName
-	cmds.AddCommand(NewCmdCreateSecret(NewSecretRecommendedCommandName, newSecretFullName, f, out))
```
Comment (deads2k) [inline pkg/oc/cli/secrets/subcommand.go]: I expect a deprecation warning, but I still expect to be able to run `oc secret create`.  Is this not doing what I think it's doing?

Comment (juanvallejo) [reply pkg/oc/cli/secrets/subcommand.go]: I believe they are still being added to the `oc secrets` tree here https://github.com/openshift/origin/pull/18093/files#diff-98f0c62da95151f788ac0edc23423802R200, but with the deprecation warning

Comment (soltysh) [reply pkg/oc/cli/secrets/subcommand.go]: Yes, you still can run those, it'll just scream after you with the deprecation.

File: pkg/oc/cli/secrets/basicauth.go [hunk 2]
```diff
@@ -69,6 +69,9 @@ func NewCmdCreateBasicAuthSecret(name, fullName string, f kcmdutil.Factory, read
 		Short:   "Create a new secret for basic authentication",
 		Long:    createBasicAuthSecretLong,
 		Example: fmt.Sprintf(createBasicAuthSecretExample, fullName, newSecretFullName, ocEditFullName),
+		PreRun: func(cmd *cobra.Command, args []string) {
```
Comment (soltysh) [inline pkg/oc/cli/secrets/basicauth.go]: I'm going to change this to `Deprecated` variable in `cobra.Command`.

**Comment by soltysh** [comment]
This is ready to go, @deads2k @juanvallejo make the honors

**Pull request merged**